includes/class-freemius.php

```diff
@@ -12920,6 +12925,92 @@ function _add_trial_notice() {
 		}
 
 		/**
+		 * Lets users/customers know that the product has an affiliate program.
+		 *
+		 * @author Leo Fajardo (@leorw)
+		 * @since  1.2.2.11
+		 *
+		 * @return bool Returns true if the notice has been added.
+		 */
+		function _add_affiliate_program_notice() {
+			if ( ! $this->is_user_admin() ) {
+				return false;
+			}
+
+			if ( ! $this->is_user_in_admin() ) {
+				return false;
+			}
+
+			// Check if the notice is already shown.
+			if ( $this->_admin_notices->has_sticky( 'affiliate_program' ) ) {
+				return false;
+			}
+
+            $last_time_notice_was_shown = $this->_storage->get( 'affiliate_program_notice_shown', false );
+            $was_notice_shown_before    = ( false !== $last_time_notice_was_shown );
+
+            /**
+             * Do not show the notice if it was already shown before or less than 30 days have passed since the initial
+             * activation with FS.
+             */
+            if ( $was_notice_shown_before ||
+                $this->_storage->install_timestamp > ( time() - ( WP_FS__TIME_24_HOURS_IN_SEC * 30 ) )
+            ) {
+                return false;
+            }
+
+            if (
+                // Product has no affiliate program.
+                ! $this->has_affiliate_program() ||
+                // User is already an affiliate.
+                is_object( $this->affiliate ) ||
+                // User has applied for an affiliate account.
+                ! empty( $this->_storage->affiliate_application_data ) ) {
+				return false;
+			}
+
+			if ( ! $this->is_paying() &&
+                FS_Plugin::AFFILIATE_MODERATION_CUSTOMERS == $this->_plugin->affiliate_moderation ) {
+			    // If the user is not a customer and the affiliate program is only for customers, don't show the notice.
+                return false;
+			}
+
+			if ( ! $this->apply_filters( 'show_affiliate_program_notice', true ) ) {

```

Comment:
@fajardoleo same about this, if the developer specified not to show the notice, it should be checked before the logic that checks if the notice was visible before.